clustermesh/certs.go

```diff
@@ -183,25 +183,21 @@ func (k *K8sClusterMesh) deleteCertificates(ctx context.Context) error {
 }
 
 func (k *K8sClusterMesh) installCertificates(ctx context.Context) error {
-	err := k.certManager.LoadCAFromK8s(ctx)
+
+	caSecret, err := k.certManager.GetCASecret(ctx)
 	if err != nil {
-		if !k.params.CreateCA {
-			k.Log("❌ Cilium CA not found: %s", err)
-			return err
-		}
-
-		k.Log("🔑 Generating CA...")
-		if err := k.certManager.GenerateCA(); err != nil {
-			return fmt.Errorf("unable to generate CA: %w", err)
-		}
-
-		if err := k.certManager.StoreCAInK8s(ctx); err != nil {
-			return fmt.Errorf("unable to store CA in secret: %w", err)
-		}
-	} else {
-		k.Log("🔑 Found existing CA in secret %s", defaults.CASecretName)
+		k.Log("❌ Unable to get Cilium CA: %s", err)
+		return err
 	}
 
+	err = k.certManager.LoadCAFromK8s(ctx, caSecret)
+	if err != nil {
+		k.Log("❌ Unable to load Cilium CA: %s", err)
+		return err
+	}
+
+	k.Log("🔑 Found CA in secret %s", defaults.CASecretName)
```

Comment:
Since the function that loads the secret doesn't take this as a param, sooner or later this log line is going to break and be out of date. It would be better to accept `defaults.CASecretName` as a param to `GetCASecret`

Reply:
Updated GetCASecret to include this parameter